graphql_compiler/compiler/ir_lowering_match.py

```diff
@@ -8,19 +8,30 @@
 us to convert this Expression into other Expressions, using data already present in the IR,
 to simplify the final code generation step.
 """
+from collections import namedtuple
 import funcy.py2 as funcy
+import itertools
 import six
 
-from .blocks import Backtrack, CoerceType, Filter, MarkLocation, QueryRoot, Traverse
-from .expressions import (BinaryComposition, ContextField, ContextFieldExistence, FalseLiteral,
-                          Literal, LocalField, TernaryConditional, TrueLiteral)
+from .blocks import (Backtrack, CoerceType, ConstructResult,
+                     Filter, MarkLocation, QueryRoot, Traverse)
+from .expressions import (BinaryComposition, ContextField, ContextFieldExistence,
+                          FalseLiteral, FoldedOutputContextField, Literal, LocalField,
+                          NullLiteral, OutputContextField, TernaryConditional,
+                          TrueLiteral, UnaryTransformation, ZeroLiteral)
 from .ir_lowering_common import (lower_context_field_existence, merge_consecutive_filter_clauses,
                                  optimize_boolean_expression_comparisons)
 from .ir_sanity_checks import sanity_check_ir_blocks_from_frontend
-from .match_query import MatchStep, convert_to_match_query
+from .match_query import MatchQuery, MatchStep, convert_to_match_query
 from .workarounds import orientdb_class_with_while, orientdb_eval_scheduling
 
 
+###
+# A CompoundMatchQuery is a representation of several MatchQuery objects containing
+#   - match_queries: a list MatchQuery objects
+CompoundMatchQuery = namedtuple('CompoundMatchQuery', ('match_queries'))
```

Comment:
Seems unnecessary, why not just have `convert_optional_traversal_to_compound_match_query` return `list[MatchQuery]` ?

Reply:
You can fight this one out with @obi1kenobi  ;)